madeofhuman/maintenance-react
# #161052234 improve documentation

#### What does this PR do?
adds comprehensive documentation in form of a descriptive readme page and JSDocs for all classes and functions.

#### Description of Task to be completed?
- create a comprehensive readme file
- add JSDoc comments to the code

#### How should this be manually tested?
- visit https://github.com/madeofhuman/maintenance-react to see a thorough readme file
- go through the code to see proper JSDoc comments

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161052234](https://www.pivotaltracker.com/story/show/161052234)

#### Screenshots
N/A

#### Questions:
N/A

[Delivers #161052234]
